scripts/actions/mobskills/astral_flow.lua

```diff
@@ -23,11 +12,7 @@ mobskillObject.onMobWeaponSkill = function(target, mob, skill)
     skill:setMsg(xi.msg.basic.USES)
 
     local mobID  = mob:getID()
-    local avatar = mobID + 2 -- default offset
-
-    if avatarOffsets[mobID] then
-        avatar = mobID + avatarOffsets[mobID]
-    end
+    local avatar = mobID + mob:getMobMod(xi.mobMod.ASTRAL_PET_OFFSET)
```

Comment:
Instead of toucking core, why not check the mobmod value here and default it to 2 if 0?
```lua
local offset = mob:getMobMod(xi.mobMod.ASTRAL_PET_OFFSET) > 0 and mob:getMobMod(xi.mobMod.ASTRAL_PET_OFFSET) or 2
local avatar = mobID + offset
```